src/extension.js

```diff
@@ -0,0 +1,263 @@
+      let link = results.items[results.focusedIndex];
+      lastNavigation.lastQueryUrl = location.href;
+      lastNavigation.lastFocusedIndex = results.focusedIndex;
+      that.options.local.save();
+      link.click();
+    });
+
+    this.register(options.navigateNewTabKey, () => {
+      let link = results.items[results.focusedIndex];
+      window.open(link.href);
+    });
+  },
+
+  initCommonGoogleSearchNavigation: function() {
+    let options = this.options.sync.values;
+
+    this.register(options.focusSearchInput, () => {
+      let searchInput = document.getElementById('lst-ib');
+      searchInput.focus();
+      searchInput.select();
+    });
+
+    let tabs = [
+      [options.navigateSearchTab, '//a[contains(@class, \'q qs\') and not (contains(@href, \'&tbm=\')) and not (contains(@href, \'maps.google.\'))]'],
+      [options.navigateImagesTab, '//a[contains(@class, \'q qs\') and (contains(@href, \'&tbm=isch\'))]'],
+      [options.navigateVideosTab, '//a[contains(@class, \'q qs\') and (contains(@href, \'&tbm=vid\'))]'],
+      [options.navigateMapsTab, '//a[contains(@class, \'q qs\') and (contains(@href, \'maps.google.\'))]'],
+      [options.navigateNewsTab, '//a[contains(@class, \'q qs\') and (contains(@href, \'&tbm=nws\'))]'],
+      [options.navigatePreviousResultPage, "//a[@id='pnprev']"],
+      [options.navigateNextResultPage, "//a[@id='pnnext']"]
+    ];
+
+    for (let i = 0; i < tabs.length; i++) {
+      let tabCommand = tabs[i];
+      this.register(tabCommand[0], () => {
+        let node = getElementByXpath(tabCommand[1]);
+
+        if (node !== null) {
+          location.href = node.href;
+        }
+      });
+    }
+  },
+
+  register: function(shortcut, callback) {
+    key(shortcut, function(event) {
+      callback();
+
+      if (event !== null) {
+        event.stopPropagation();
+        event.preventDefault();
+      }
+
+      return false;
+    });
+  }
+};
+
+/**
+ * @param {StorageArea} storage The storage area to which this section will write.
+ * @param {Object} defaultValues The default options.
+ * @constructor
+ */
+function OptionSection(storage, defaultValues) {
+  this.storage = storage;
+  this.values = defaultValues;
+
+  this.load = function() {
+    return new Promise((resolve) => {
+      this.storage.get(
+        this.values,
+        (values) => {
+          if (!chrome.runtime.lastError) {
+            this.values = values;
+          }
+          resolve();
+        }
+      );
+    });
+  };
+
+  this.save = function() {
+    return new Promise((resolve, reject) => {
+      this.storage.set(
+        this.values,
+        () => {
+          if (chrome.runtime.lastError) {
+            reject();
+          }
+          else {
+            resolve();
+          }
+        }
+      )
+    });
+  };
+}
+
+/**
+ * @param {NodeList|Node[]} nodes The collection of the anchor nodes representing search result to move focus for.
+ * @constructor
+ */
+function SearchResults(nodes) {
```

Comment:
const SearchResults = (nodes) => {

Reply:
I can't do that. In such case `SearchResults` wouldn't be a constructor.

Reply:
Ah, you're right. Keep it this way then.